### ISSUE TITLE
ts-server gets high latency (about 5 seconds per action) for ts/js language feature

### ISSUE THREAD

**Issue by Arichy** [microsoft/vscode#159329]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.70.2
- OS Version: macos 12.5.1

I made a simple demo to reproduce this problem: https://github.com/Arichy/minitest
Steps to Reproduce:

1. git clone https://github.com/Arichy/minitest.git
2. open the repo in vscode
3. yarn install
4. goto `src/index.tsx` and type whatever ts/js code as you're really coding

It seems ts throw an error: Expression produces a union type that is too complex to represent.
But I don't know why ts throws the error. I only know the `memo` without generic in `Comp.ts` and the collection of custom elements declaration in `react-app-env.d.ts` cause the error together after trying many times.
 
The problem acts like this:

![Kapture 2022-08-27 at 00 56 58](https://user-images.githubusercontent.com/29599723/186955143-7dd14e3f-d9d9-4b4b-ab19-36050fbd29b0.gif)

As you see, ts language features such as autocompletion, goto definition, method hint are all having very high latency. It will be more serious in my practical business project.

**Comment by mjbvz** [comment]
Does this reproduce in the [latest VS Code insiders build](https://code.visualstudio.com/insiders/) with all [extensions disabled](https://github.com/microsoft/vscode/wiki/Performance-Issues#run-with-extensions-disabled)?

You example shows you have IntelliCode enabled which easily explain this

**Comment by Arichy** [comment]
> Does this reproduce in the [latest VS Code insiders build](https://code.visualstudio.com/insiders/) with all [extensions disabled](https://github.com/microsoft/vscode/wiki/Performance-Issues#run-with-extensions-disabled)?
> 
> You example shows you have IntelliCode enabled which easily explain this

Yes. It reproduces in the latest vscode insider: 1.71.0 (commit id: 35b971c92d210face8c446a1c6f1e470ad2bcb54) with all extensions disabled.
This issue is about vscode's builtin ts-server, so I cannot disable ts features.
